src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -2555,22 +2556,71 @@ def _extract_token_timestamps(self, generate_outputs, alignment_heads, time_prec
         # of shape (batch size, num selected, output length, input length).
         weights = torch.stack([cross_attentions[l][:, h] for l, h in alignment_heads])
         weights = weights.permute([1, 0, 2, 3])
+
+        if "beam_indices" in generate_outputs:
+            # If beam search has been used, the output sequences can have been generated for more timesteps than their sequence_lengths
+            # Since the beam search strategy choose the most probable sequences at the end of the search
+            # In that case, we make sure that the weights have the right output_length
+            weight_length = (generate_outputs.beam_indices != -1).sum(-1).max()
+            weights = weights[:, :, :weight_length]
+
+            # If beam indice is still -1, it means that the associated token id is EOS
+            beam_indices = generate_outputs.beam_indices[:, :weight_length]
+            beam_indices = beam_indices.masked_fill(beam_indices == -1, 0)
+
+            # Select the cross attention from the right beam for each output sequences
+            weights = torch.stack(
+                [
+                    torch.index_select(weights[:, :, i, :], dim=0, index=beam_indices[:, i])
+                    for i in range(beam_indices.shape[1])
+                ],
+                dim=2,
+            )
+
+        timestamps = torch.zeros_like(generate_outputs.sequences, dtype=torch.float32)
+        batch_size = timestamps.shape[0]
```

Comment:
Could you check that this makes sense @gante ?

Basically, this reconciles `generate_outputs.cross_attentions` with `generate_outputs.sequences` thanks to `generate_outputs.beam_indices`.